src/components/modal/LoginCardModal.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect } from 'react';
+import { useAuth } from '../../context/auth/auth-context';
+import '../css/products-page.css';
+import { useModal } from '../../context/modal/modal-context';
+const LoginCardModal = () => {
+	const { isLogin, isLoading, error, toggleIsLogin, signUpHandler, loginHandler, logoutHandler, errorHandler } = useAuth();
+	const { toggleModal } = useModal();
+
+	const defaultText = {
+		email: '',
+		password: '',
+		confirmPassword: '',
+		emailError: false,
+		passWordError: false,
+		confirmPasswordError: false,
+	};
+
+	const [textFields, setTextFields] = useState(defaultText);
+
+	const setUserDetails = (e) => {
+		console.log(e.target.name);
+		errorHandler(false, '');
+		setTextFields({
+			...textFields,
+			[e.target.name]: e.target.value,
+		});
+	};
+
+	const signupChecker = () => {
+		const { email, password, confirmPassword } = textFields;
+
+		email.length !== 0
+			? email
+					.toLowerCase()
+					.match(
+						/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+					)
+				? password.length !== 0
+					? confirmPassword.length !== 0
+						? password === confirmPassword
+							? signUpHandler({ email: email, password: password })
+							: (setTextFields({
+									...textFields,
+									confirmPasswordError: true,
+							  }),
+							  errorHandler(true, 'Password and Confirm Password must be same'))
+						: (setTextFields({ ...textFields, confirmPasswordError: true }), errorHandler(true, 'Confirm Password is required'))
+					: (errorHandler(true, 'Password is required'), setTextFields({ ...textFields, passWordError: true }))
+				: (errorHandler(true, 'Email is invalid'), setTextFields({ ...textFields, emailError: true }))
+			: (setTextFields({ ...textFields, emailError: true }), errorHandler(true, 'Email is required'));
+	};
+
+	const loginChecker = async () => {
+		const { email, password } = textFields;
+
+		return email.length !== 0
+			? email
+					.toLowerCase()
+					.match(
+						/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+					)
+				? password.length !== 0
+					? loginHandler({ email: email, password: password })
+					: (errorHandler(true, 'Password is required'), setTextFields({ ...textFields, passWordError: true }))
+				: (errorHandler(true, 'Email is invalid'), setTextFields({ ...textFields, emailError: true }))
+			: (setTextFields({ ...textFields, emailError: true }), errorHandler(true, 'Email is required'));
+	};
+
+	return (
+		<div className='modal-card'>
+			<div className='flex-column align-items-center space-evenly'>
+				<h1>{isLogin ? 'Login' : 'Sign Up'}</h1>
+				<div className='input-group'>
+					<InputField
+						labelText='Email Address'
```

Comment:
Suggestion if possible to add the required tag in the inputField for login and password.

Reply:
I've already added logic to check all the possibilities of input.